### ISSUE TITLE
Taint not found inside "if" statement

### ISSUE THREAD

**Issue by mmcev106** [vimeo/psalm#7950]
Should this taint be reported?

https://psalm.dev/r/8c624220b8

**Comment by AndrolGenhald** [comment]
Yes, but currently the type of `$a` is lost once the `if` scope ends. I have a branch I've done some work on to fix this, but I haven't had much time the past few months to finish it and get it working. Hopefully I'll have more time in a month or two, I'll add this as a test case to make sure it works.

**Comment by mmcev106** [comment]
Thanks for the info!  I'm curious, which branch would that be?

**Comment by AndrolGenhald** [comment]
I think it's still local atm, it's failing a lot of tests and still needs quite a bit of work. I started working on loops and ifs after I got stuck on the same issue in #7688, since the variable not being defined outside of the if scope caused problems with my TryAnalyzer improvements.

**Comment by mmcev106** [comment]
@AndrolGenhald, any updates here?  If not, would it make sense for that branch to be made public so that others might be able to help?

**Comment by orklah** [comment]
The branch is [here ](https://github.com/AndrolGenhald/psalm/tree/try-analyzer-rewrite) if anyone want to look at it
